File: tasks/https_get/ruby/task.rb
require 'net/http'

uri = URI('https://example.com/my-path')

request = Net::HTTP::Get.new(uri.request_uri)
request['My-Header-Key'] = 'my-header-value'
request.body = "my-body-text"

response = Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') do |http|
  http.request(request)
end

puts response.body

File: tasks/https_get/java/Task.java
// This sample uses the Apache HTTP client from HTTP Components.
// See http://hc.apache.org/httpcomponents-client-ga/
import java.net.URI;
import org.apache.http.HttpEntity;
import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.client.utils.URIBuilder;
import org.apache.http.impl.client.HttpClients;
import org.apache.http.util.EntityUtils;

public class JavaSample 
{
    public static void main(String[] args) 
    {
        HttpClient httpclient = HttpClients.createDefault();

        try
        {
            URIBuilder builder = new URIBuilder("https://example.com/my-path");
            URI uri = builder.build();
            HttpGet request = new HttpGet(uri);
            request.setHeader("My-Header-Key", "my-header-value");
            StringEntity reqEntity = new StringEntity("my-body-text");
            request.setEntity(reqEntity);
            HttpResponse response = httpclient.execute(request);
            HttpEntity entity = response.getEntity();

            if (entity != null) 
            {
                System.out.println(EntityUtils.toString(entity));
            }
        }
        catch (Exception e)
        {
            System.out.println(e.getMessage());
        }
    }
}
